runtime/vm/createramclass.cpp

```diff
@@ -3531,10 +3531,24 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 		}
 		
 		if (J9_ARE_ALL_BITS_SET(romClass->modifiers, J9AccValueType)) {
+			UDATA instanceSize = result->totalInstanceSize;
 			classFlags |= J9ClassIsValueType;
-			if ((result->totalInstanceSize <= javaVM->valueFlatteningThreshold) && !J9ROMCLASS_IS_CONTENDED(romClass)) {
-				Trc_VM_CreateRAMClassFromROMClass_valueTypeIsFlattened(vmThread, J9UTF8_LENGTH(className), J9UTF8_DATA(className), result);
-				classFlags |= J9ClassIsFlattened;
+			if ((instanceSize <= javaVM->valueFlatteningThreshold)
+				&& !J9ROMCLASS_IS_CONTENDED(romClass)
+			) {
+				if (J9ROMCLASS_IS_ATOMIC(romClass)) {
+					UDATA unpaddedInstanceSize = J9_ARE_ALL_BITS_SET(valueTypeFlags, J9ClassRequiresPrePadding) ? (instanceSize - sizeof(U_32)) : instanceSize;
```

Comment:
Please write this as an assign instanceSize followed by if/subtract.

Reply:
Fixed.